client/network/sync/src/engine.rs

```diff
@@ -698,8 +686,24 @@ where
 					}
 				},
 				ToServiceCommand::AnnounceBlock(hash, data) => self.announce_block(hash, data),
-				ToServiceCommand::NewBestBlockImported(hash, number) =>
-					self.new_best_block_imported(hash, number),
+				ToServiceCommand::NewBestBlockImported(hash, number) => {
+					log::debug!(target: "sync", "New best block imported {:?}/#{}", hash, number);
+
+					self.chain_sync.update_chain_info(&hash, number);
+					while let Poll::Pending = self
```

Comment:
I'm not quite sure why do we need a loop here (isn't it a busy loop after all?), as there were no such a thing in the previous code.

Reply:
`SyncingEngine` is lagging behind in terms of refactoring and still heavily uses polling. There is work that tries to address this (https://github.com/paritytech/substrate/pull/14675) and in time this call can be converted to a simple `.await`.

As far as these changes are concerned, updating the handshake is now a future because the code sends the command to `Notifications` directly over a bounded command channel. The channel is not busy and right now expects one message every six seconds so I don't believe this to be a grave concern. I will add a TODO to remove this ugly loop once `SyncingEngine` is ready.